crates/subspace-farmer/src/utils/parity_db_store.rs

```diff
@@ -1,21 +1,28 @@
 use parity_db::{ColumnOptions, Db, Options};
 use std::error::Error;
 use std::fmt;
+use std::fmt::Debug;
 use std::marker::PhantomData;
 use std::path::Path;
 use std::sync::Arc;
-use subspace_core_primitives::Piece;
-use subspace_networking::libp2p::kad::record::Key;
 use tracing::{debug, trace, warn};
 
 /// Generic key value store with ParityDB backend and iteration support
 #[derive(Clone)]
-pub struct ParityDbStore {
+pub struct ParityDbStore<StoreKey, StoreValue> {
     // Parity DB instance
     db: Arc<Db>,
+
+    // Type marker
+    marker: PhantomData<(StoreKey, StoreValue)>,
 }
 
-impl ParityDbStore {
+impl<StoreKey, StoreValue> ParityDbStore<StoreKey, StoreValue>
+where
+    StoreKey: AsRef<[u8]> + From<Vec<u8>> + Debug,
+    StoreValue: TryFrom<Vec<u8>>,
+    StoreValue::Error: Debug,
```

Comment:
I'm so happy this now compiles, it didn't in past Rust versions and it was necessary to write `<StoreValue as TryFrom<Vec<u8>>::Error` instead